tests/foreman/destructive/test_clone.py

```diff
@@ -0,0 +1,111 @@
+pytestmark = pytest.mark.destructive
+
+
+@pytest.mark.parametrize("sat_ready_rhel", [8], indirect=True)
+@pytest.mark.parametrize('backup_type', ['online', 'offline'])
+@pytest.mark.parametrize('skip_pulp_content', [True, False], ids=['skip_pulp', 'no_skip_pulp'])
+def test_positive_clone_backup(target_sat, sat_ready_rhel, backup_type, skip_pulp_content):
+    """Make an online/offline backup with/without pulp data of Satellite and clone it (restore it).
+
+    :id: 5b9182d5-6789-4d2c-bcc3-6641b96ab277
+
+    :steps:
+        1. Create an online/offline backup with/without pulp data
+        2. Copy it to target rhel
+        3. Install satellite-clone
+        4. Run satellite clone on target rhel
+
+    :expectedresult:
+        1. Satellite-clone ran successfully
+
+    :parametrized: yes
+    """
+    rhel_version = sat_ready_rhel._v_major
+    sat_version = target_sat.version
+
+    # SATELLITE PART - SOURCE SERVER
+    # Enabling and starting services
+    assert target_sat.cli.Service.enable().status == 0
+    assert target_sat.cli.Service.start().status == 0
+
+    # Doing backup
+    backup_result = target_sat.cli.Backup.run_backup(
+        backup_dir='/var/backup',
+        backup_type=backup_type,
+        options={'assumeyes': True, 'plaintext': True, 'skip-pulp-content': skip_pulp_content},
+    )
+    assert backup_result.status == 0
+    sat_backup_dir = backup_result.stdout.strip().split()[-2]
+
+    if skip_pulp_content:
+        # Copying satellite pulp data to target RHEL
+        assert sat_ready_rhel.execute('mkdir -p /var/lib/pulp').status == 0
+        assert (
+            target_sat.execute(
+                f'''sshpass -p "{SSH_PASS}" scp -o StrictHostKeyChecking=no \
+                -r /var/lib/pulp root@{sat_ready_rhel.hostname}:/var/lib/pulp/pulp'''
+            ).status
+            == 0
+        )
+    # Copying satellite backup to target RHEL
+    assert (
+        target_sat.execute(
+            f'''sshpass -p "{SSH_PASS}" scp -o StrictHostKeyChecking=no \
+            -r {sat_backup_dir} root@{sat_ready_rhel.hostname}:/backup/'''
+        ).status
+        == 0
+    )
+    assert target_sat.cli.Service.stop().status == 0
+    assert target_sat.cli.Service.disable().status == 0
+
+    # RHEL PART - TARGET SERVER
+    assert sat_ready_rhel.execute('ls /backup/config_files.tar.gz').status == 0
+    # Registering to cdn
+    sat_ready_rhel.register_to_cdn()
+    # Disabling repositories
+    assert sat_ready_rhel.execute('subscription-manager repos --disable=*').status == 0
+    # Getting satellite maintenace repo
+    sat_ready_rhel.download_repofile(product='satellite', release=sat_version)
+    # Enabling repositories
+    for repo in getattr(constants, f"OHSNAP_RHEL{rhel_version}_REPOS"):
+        sat_ready_rhel.enable_repo(repo, force=True)
+    # Enabling satellite module
+    assert sat_ready_rhel.execute(f'dnf module enable -y satellite:el{rhel_version}').status == 0
```

Comment:
It would be better to move these steps to the `sat_ready_rhel` fixture where it has all required specs and repos configured, what do you think?

Reply:
I think that this fixture should be generic so it can also be used in other tests. This also means that I personally would prefer to leave this RHEL setup in the test, because i.e. sometimes you could use this fixture to install a satellite or sometimes you would want to use this just to clone a satellite that requires different repos.

Reply:
I would definitely not put the setup into the `sat_ready_rhel` fixture. Instead, I would add a new fixture/set of fixtures that are dependent on `sat_ready_rhel`, to perform this setup.
However, I see this as a TODO for future satellite-clone test expansions.

Reply:
Yes, you're right, I actually meant to include this steps as part of fixture, and not particularly `sat_ready_rhel` fixture